doc/api/string_decoder.md

```diff
@@ -17,23 +23,49 @@ const euro = Buffer.from([0xE2, 0x82, 0xAC]);
 console.log(decoder.write(euro));
 ```
 
-## Class: StringDecoder
+When a `Buffer` instance is written to the `StringDecoder` instance, an
+internal buffer is used to ensure that the decoded string does not contain
+any partial multibyte-characters. These are held in the buffer until the
+next call to `stringDecoder.write()` or until `stringDecoder.end()` is called.
+
+For following example, for instance, writes the three bytes for the European
+euro symbol over three separate operations:
+
+```js
+const StringDecoder = require('string_decoder').StringDecoder;
+const decoder = new StringDecoder('utf8');
+
+decoder.write(Buffer.from([0xE2]));
+decoder.write(Buffer.from([0x82]));
+console.log(decoder.write(Buffer.from([0xAC])));
+```
+
+## Class: new StringDecoder([encoding])
 <!-- YAML
 added: v0.1.99
 -->
 
-Accepts a single argument, `encoding` which defaults to `'utf8'`.
+* `encoding` {string} The character encoding the `StringDecoder` will use.
+  Defaults to `'utf8'`.
+
+Creates a new `StringDecoder` instance.
 
-### decoder.end()
+### stringDecoder.end()
 <!-- YAML
 added: v0.9.3
 -->
 
-Returns any trailing bytes that were left in the buffer.
+Returns any trailing bytes that were left in the buffer as a string. Bytes that
```

Comment:
`Returns any remaining input`? `trailing` seems to imply that it’s intentional that there are incomplete chacters